java/src/main/java/com/spotify/voyager/jni/Index.java

```diff
@@ -277,6 +277,15 @@ public void close() throws IOException {
     nativeDestructor();
   }
 
+  @Override
+  protected void finalize() throws Throwable {
```

Comment:
this is called by the JVM on garbage collection so the destructor will be called even if `close` isn't called